cluster_zonal.tf

```diff
@@ -26,8 +26,8 @@ resource "google_container_cluster" "zonal_primary" {
   zone             = "${var.zones[0]}"
   additional_zones = ["${slice(var.zones,1,length(var.zones))}"]
 
-  network            = "${data.google_compute_network.gke_network.self_link}"
-  subnetwork         = "${data.google_compute_subnetwork.gke_subnetwork.self_link}"
+  network            = "projects/${var.project_id}/global/networks/${data.google_compute_network.gke_network.name}"
```

Comment:
Isn't that equivalent to the self_link?

The issue with formatting it like this is if the cluster is being launched on a shared VPC, in which case the network is in a *different* project.

Reply:
_self_link_ adds `https://www.googleapis.com/compute/v1` before the `projects/[..]`, hence the permadiff since terraform sees the current path as the one starting from `projects/[..]` and tries to update it.

I'll take a look at the shared VPC case and will come with a solution/an idea hopefully. :-)

Reply:
@morgante I reused the 'data.google_compute_network` to get the project_id, which takes the shared VPC into account right now. Proposed in 8f96807c4a211ed29d2c6d60b343d87ae1c4cfba

Reply:
One alternative idea: what if we used the self_link but used `replace` to strip the URL prefix? https://www.terraform.io/docs/configuration/interpolation.html#replace-string-search-replace-

Reply:
Sounds good.
As simple as `network = "${replace(data.google_compute_network.gke_network.self_link, "https://www.googleapis.com/compute/v1/", "")}"` ?

Reply:
Added in e997e6604fce1e9d9e77f89c483792f26cc934de. :-)